crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -778,7 +778,7 @@ where
                 if self.settings.rules.any_enabled(&[
                     Rule::MutableDataclassDefault,
                     Rule::FunctionCallInDataclassDefaultArgument,
-                ]) && ruff::rules::is_dataclass(self, decorator_list)
+                ]) && ruff::rules::is_dataclass(&self.model, decorator_list)
```

Comment:
Can we rename `model` here to `semantic_model`. I just reviewed a PR and was so confused by `self.model`. I was like, what's this? Where is this model coming from. It being `semantic_model` would make it clear (or `semantic` if you want to keep it short`)